src/JustEat.StatsD/Buffered/StatsDMessage.cs

```diff
@@ -1,31 +1,39 @@
+using System.Collections.Generic;
+
 namespace JustEat.StatsD.Buffered
 {
     internal readonly struct StatsDMessage
     {
         public readonly string StatBucket;
         public readonly double Magnitude;
         public readonly StatsDMessageKind MessageKind;
+        public readonly Operation Operation;
+        public readonly IDictionary<string, string>? Tags;
 
-        private StatsDMessage(string statBucket, double magnitude, StatsDMessageKind messageKind)
+        private StatsDMessage(string statBucket, double magnitude, StatsDMessageKind messageKind,
+            IDictionary<string, string>? tags, Operation operation = default)
```

Comment:
Nit: one parameter per line if not all on the same line please.

```suggestion
        private StatsDMessage(
            string statBucket,
            double magnitude,
            StatsDMessageKind messageKind,
            IDictionary<string, string>? tags,
            Operation operation = default)
```